lib/dataset/except.cpp

```diff
@@ -27,37 +27,51 @@ void throw_mismatch_error(const dataset::Dataset &expected,
                      to_string(actual) + '.' + optional_message);
 }
 
-CoordMismatchError::CoordMismatchError(const Dim dim, const Variable &expected,
-                                       const Variable &actual)
-    : DatasetError{"Mismatch in coordinate '" + to_string(dim) +
-                   "', expected\n" + format_variable(expected) + ", got\n" +
-                   format_variable(actual)} {}
+namespace {
+auto format_coord_mismatch_message(const Dim dim, const Variable &a,
+                                   const Variable &b,
+                                   const std::string_view opname) {
+  std::string message = "Mismatch in coordinate '" + to_string(dim);
+  if (!opname.empty())
+    message += "' in operation '" + std::string(opname);
+  message += "':\n" + format_variable(a) + "\nvs\n" + format_variable(b);
```

Comment:
Could be nice to avoid the extra line for the "vs"? The output would be easier to compare by eye if it was on adjacent lines?

Reply:
That is what is was like before. I found it hard to see the old 'got' and had to pay close attention to see that it was showing a comparison.
But I can change it if you want.

Reply:
Maybe we can reformulate to put something with this meaning in the line before? Or prefix the lines? Something like:

```
Mismatch in coordinate 'x':
a: <formatted variable>
b: <formatted variable>
```

Reply:
The a, b labels might be a bit confusing. My first reaction was 'are they coord names?' (before reading the actual message...). What about `1:` and `2:`?

Reply:
Sounds good.